applications/CompressiblePotentialFlowApplication/custom_processes/define_2d_wake_process.cpp

```diff
@@ -0,0 +1,326 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Inigo Lopez
+//
+
+// Project includes
+#include "define_2d_wake_process.h"
+#include "utilities/variable_utils.h"
+#include "utilities/openmp_utils.h"
+
+namespace Kratos {
+
+// Constructor for Define2DWakeProcess Process
+Define2DWakeProcess::Define2DWakeProcess(ModelPart& rBodyModelPart, const double Tolerance)
+    : Process(), mrBodyModelPart(rBodyModelPart), mTolerance(Tolerance)
+{
+    ModelPart& root_model_part = mrBodyModelPart.GetRootModelPart();
+    root_model_part.CreateSubModelPart("trailing_edge_sub_model_part");
```

Comment:
What do you think about checking if the trailing_edge_sub_model_part already exists, and removing it or reading the existing one in case it's already defined?

Do you think this is the place to do so? Maybe we can also store a list of pointers instead of creating a submodelpart, which might give a better performance. What is your opinion @rubenzorrilla ?

My only concern is when running again the same model that has data already loaded (for instance in a MonteCarlo analysis with the potential flow), and we try to create a model part that already exists. Probably this is also happening when running FSI cases?

Reply:
For me makes sense to check if it already exists. If so it does, I'd remove all the nodes, elements and conditions from it and you reuse this empty model part to store the wake. If not, you create it as you are doing right now.

If you are aiming to use this in a MLMC analysis, it will for sure throw an error when trying to create a model part with an already existing name.

For the FSI we don't have this problem (so far) but I've faced similar ones when dealing with the intersections skin in embedded formulations.

Reply:
@marcnunezc good point raising this issue here. You are right, I will add the check here.

Just some comments:

- The current implementation already reads the existing one if it already exists, for example in: https://github.com/KratosMultiphysics/Kratos/blob/beeea49b84aaccf71cab9171fdc8f4e1d98646e1/applications/CompressiblePotentialFlowApplication/custom_processes/define_2d_wake_process.cpp#L245-L246

- I like the idea of storing a list of pointers instead of creating a submodelpart (@philbucher what do you think about this?). I don't think it will have a large impact in the performance when using it for the trailing edge submodelpart because it only has a few elements, but I will definitely look into this when creating the wake submodelpart (specially in 3D).

- Since the creation of the submodelpart happens at the constructor of the process, we don't have the issue of having an already existing submodelpart in FSI, because we only call the function ExecuteInitialize in each coupling iteration. However, as @rubenzorrilla mentioned, for FSI we first need to clear the marking of all nodes, elements and conditions existing in the submodelpart from the previous iteration and then we need to remove all nodes, elements and conditions.

Reply:
I don't agree with the idea of creating a list of pointers since the overhead of having a model part is minimum and gives you much more flexibility. Note that having it makes it possible to manipulate the wake container from outside this process through the model part API.

Reply:
That's a good point. Thanks for the hint @rubenzorrilla, we will stay with the model part then.